src/tools/convert/topology_helper.py

```diff
@@ -58,7 +58,12 @@ def FillFileBuffer(fname, filebuffer):
     for line in f:
 	if "include" in line and not line[0]==';':
 	    name=(line.split()[1]).strip('\"')
-	    FillFileBuffer(name, filebuffer)
+            try: 
+ 	        FillFileBuffer(name, filebuffer)
+            except IOError:
+                #need to use relative path
+                name = fname.split('/')[0] + '/' + name
```

Comment:
Gromacs topology files give the path of other included topology files as filename or subdirectory/filename, relative to the cwd, but sometimes the file given simply as filename is actually in the subdirectory of the file it was included from, not in the cwd. Gromacs can handle this, espresso++ couldn't.

In Gromacs-format files I've only ever seen a relative path one subdirectory deep, hence the `fname.split('/')[0]`. I don't actually know how gromacs handles deeper directory trees. But it's probable it handles them the same way as your suggestion `os.path.join(os.path.dirname(fname), name)` so let's switch to that.

Plus it will work assuming we ever try to support Windows ;)